src/app/componenets/video/video.component.ts

```diff
@@ -0,0 +1,49 @@
+import { ChangeDetectorRef, Component, ElementRef, HostListener, Input, OnInit, ViewChild } from '@angular/core';
+
+@Component({
+  template: ` <div #youTubePlayer class="youtube">
+    <youtube-player
+      [videoId]="videoId"
+      [playerVars]="playerConfig"
+      suggestedQuality="highres"
+      [width]="videoWidth"
+      [height]="videoHeight"
+    >
+    </youtube-player>
+  </div>`,
+  selector: 'youtube',
+  styleUrls: ['./video.component.scss'],
+})
+export class VideoComponent implements OnInit {
+  @ViewChild('youTubePlayer') youTubePlayer: ElementRef<HTMLDivElement>;
+  @Input() videoId: string;
+
+  protected videoHeight: number;
+  protected videoWidth: number;
+
+  private YToffset: number = 17;
+
+  protected playerConfig = {
+    controls: 1,
+    mute: 1,
+    autoplay: 1,
+  };
+
+  constructor(private changeDetectorRef: ChangeDetectorRef) {}
+
+  ngOnInit() {
+    const tag = document.createElement('script');
+    tag.src = 'https://www.youtube.com/iframe_api';
+    document.body.appendChild(tag);
```

Comment:
To jest wymagane? Biblioteka każe takie rzeczy robić? w ng nie powinno odwoływać się do globalnych zmiennych np document czy window.

Reply:
Według dokumentacji https://github.com/angular/components/tree/main/src/youtube-player#readme tak

Reply:
Staraj się unikać takich implementacji, używając tego rozwiązania:
https://medium.com/ngconf/how-to-inject-document-in-angular-7d17804430b6